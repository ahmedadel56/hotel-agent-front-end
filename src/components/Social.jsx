import React from 'react';
import { IoLogoTwitter } from 'react-icons/io';
import { FaVimeoV, FaPinterestP } from 'react-icons/fa';
import { GrFacebookOption } from 'react-icons/gr';
import { TiSocialGooglePlus } from 'react-icons/ti';

const Social = () => (
  <ul className="icons">
    <li key={1} className="social-icons">
      <button
        type="button"
        className="twitter social-button"
        label="Twitter profile"
      >
        <a
          href="https://twitter.com/jorgeriosr16"
          aria-label="Twitter account"
          target="_blank"
          rel="noreferrer noopener"
        >
          <IoLogoTwitter />
        </a>
      </button>
    </li>
    <li key={2} className="social-icons">
      <button
        type="button"
        className="facebook social-button"
        label="facebook profile"
      >
        <a
          href="https://www.linkedin.com/in/jorgeriosr/"
          aria-label="Facebook account"
          target="_blank"
          rel="noreferrer noopener"
        >
          <GrFacebookOption />
        </a>
      </button>
    </li>
    <li key={3} className="social-icons">
      <button
        type="button"
        className="google social-button"
        label="google profile"
      >
        <a
          href="https://angel.co/u/jorge-rios-7"
          aria-label="Google account"
          target="_blank"
          rel="noreferrer noopener"
        >
          <TiSocialGooglePlus />
        </a>
      </button>
    </li>
    <li key={4} className="social-icons">
      <button
        type="button"
        className="vimeo social-button"
        label="Vimeo profile"
      >
        <a
          href="https://github.com/Alexr16"
          aria-label="Vimeo account"
          target="_blank"
          rel="noreferrer noopener"
        >
          <FaVimeoV />
        </a>
      </button>
    </li>
    <li key={5} className="social-icons">
      <button
        type="button"
        className="pinterest social-button"
        label="Pinterest profile"
      >
        <a
          href="https://medium.com/@jorge19960316"
          aria-label="Pinterest account"
          target="_blank"
          rel="noreferrer noopener"
        >
          <FaPinterestP />
        </a>
      </button>
    </li>
  </ul>
);

export default Social;
