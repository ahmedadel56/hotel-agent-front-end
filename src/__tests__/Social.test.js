import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Social from '../components/Social';

it('Check if Navbar renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Social />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});
