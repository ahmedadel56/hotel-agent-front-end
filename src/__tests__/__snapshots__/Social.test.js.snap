// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Navbar renders correctly 1`] = `
<ul
  className="icons"
>
  <li
    className="social-icons"
  >
    <button
      className="twitter social-button"
      label="Twitter profile"
      type="button"
    >
      <a
        aria-label="Twitter account"
        href="https://twitter.com/jorgeriosr16"
        rel="noreferrer noopener"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M492 109.5c-17.4 7.7-36 12.9-55.6 15.3 20-12 35.4-31 42.6-53.6-18.7 11.1-39.4 19.2-61.5 23.5C399.8 75.8 374.6 64 346.8 64c-53.5 0-96.8 43.4-96.8 96.9 0 7.6.8 15 2.5 22.1-80.5-4-151.9-42.6-199.6-101.3-8.3 14.3-13.1 31-13.1 48.7 0 33.6 17.2 63.3 43.2 80.7-16-.4-31-4.8-44-12.1v1.2c0 47 33.4 86.1 77.7 95-8.1 2.2-16.7 3.4-25.5 3.4-6.2 0-12.3-.6-18.2-1.8 12.3 38.5 48.1 66.5 90.5 67.3-33.1 26-74.9 41.5-120.3 41.5-7.8 0-15.5-.5-23.1-1.4C62.8 432 113.7 448 168.3 448 346.6 448 444 300.3 444 172.2c0-4.2-.1-8.4-.3-12.5C462.6 146 479 129 492 109.5z"
          />
        </svg>
      </a>
    </button>
  </li>
  <li
    className="social-icons"
  >
    <button
      className="facebook social-button"
      label="facebook profile"
      type="button"
    >
      <a
        aria-label="Facebook account"
        href="https://www.linkedin.com/in/jorgeriosr/"
        rel="noreferrer noopener"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.94474914,22 L9.94474914,13.1657526 L7,13.1657526 L7,9.48481614 L9.94474914,9.48481614 L9.94474914,6.54006699 C9.94474914,3.49740494 11.8713513,2 14.5856738,2 C15.8857805,2 17.0033128,2.09717672 17.3287076,2.13987558 L17.3287076,5.32020466 L15.4462767,5.32094085 C13.9702212,5.32094085 13.6256856,6.02252733 13.6256856,7.05171716 L13.6256856,9.48481614 L17.306622,9.48481614 L16.5704347,13.1657526 L13.6256856,13.1657526 L13.6845806,22"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </button>
  </li>
  <li
    className="social-icons"
  >
    <button
      className="google social-button"
      label="google profile"
      type="button"
    >
      <a
        aria-label="Google account"
        href="https://angel.co/u/jorge-rios-7"
        rel="noreferrer noopener"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.9 13.5l-.7-.5c-.2-.2-.5-.4-.5-.8s.3-.7.6-1c.8-.6 1.7-1.3 1.7-2.8 0-1.5-.9-2.3-1.4-2.7h1.2l1.2-.7h-4.1c-1 0-2.4.2-3.5 1.1-.8.7-1.2 1.7-1.2 2.6 0 1.5 1.2 3.1 3.3 3.1h.6c-.1.2-.2.4-.2.7 0 .6.3 1 .6 1.3-.9.1-2.5.2-3.8.9-1.2.7-1.5 1.7-1.5 2.4 0 1.5 1.4 2.8 4.2 2.8 3.4 0 5.2-1.9 5.2-3.7 0-1.3-.8-1.9-1.7-2.7zm-2.5-2.2c-1.7 0-2.5-2.2-2.5-3.5 0-.5.1-1 .4-1.5.3-.4.9-.7 1.4-.7 1.6 0 2.5 2.2 2.5 3.6 0 .4 0 1-.5 1.4-.3.4-.9.7-1.3.7zm0 7.9c-2.1 0-3.5-1-3.5-2.4s1.3-1.9 1.7-2c.8-.3 1.9-.3 2.1-.3h.5c1.5 1.1 2.1 1.6 2.1 2.6 0 1.2-1 2.1-2.9 2.1zM17 12h-2v-1h2v-1.9l1-.1v2h2v1h-2v2h-1z"
          />
        </svg>
      </a>
    </button>
  </li>
  <li
    className="social-icons"
  >
    <button
      className="vimeo social-button"
      label="Vimeo profile"
      type="button"
    >
      <a
        aria-label="Vimeo account"
        href="https://github.com/Alexr16"
        rel="noreferrer noopener"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M447.8 153.6c-2 43.6-32.4 103.3-91.4 179.1-60.9 79.2-112.4 118.8-154.6 118.8-26.1 0-48.2-24.1-66.3-72.3C100.3 250 85.3 174.3 56.2 174.3c-3.4 0-15.1 7.1-35.2 21.1L0 168.2c51.6-45.3 100.9-95.7 131.8-98.5 34.9-3.4 56.3 20.5 64.4 71.5 28.7 181.5 41.4 208.9 93.6 126.7 18.7-29.6 28.8-52.1 30.2-67.6 4.8-45.9-35.8-42.8-63.3-31 22-72.1 64.1-107.1 126.2-105.1 45.8 1.2 67.5 31.1 64.9 89.4z"
          />
        </svg>
      </a>
    </button>
  </li>
  <li
    className="social-icons"
  >
    <button
      className="pinterest social-button"
      label="Pinterest profile"
      type="button"
    >
      <a
        aria-label="Pinterest account"
        href="https://medium.com/@jorge19960316"
        rel="noreferrer noopener"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M204 6.5C101.4 6.5 0 74.9 0 185.6 0 256 39.6 296 63.6 296c9.9 0 15.6-27.6 15.6-35.4 0-9.3-23.7-29.1-23.7-67.8 0-80.4 61.2-137.4 140.4-137.4 68.1 0 118.5 38.7 118.5 109.8 0 53.1-21.3 152.7-90.3 152.7-24.9 0-46.2-18-46.2-43.8 0-37.8 26.4-74.4 26.4-113.4 0-66.2-93.9-54.2-93.9 25.8 0 16.8 2.1 35.4 9.6 50.7-13.8 59.4-42 147.9-42 209.1 0 18.9 2.7 37.5 4.5 56.4 3.4 3.8 1.7 3.4 6.9 1.5 50.4-69 48.6-82.5 71.4-172.8 12.3 23.4 44.1 36 69.3 36 106.2 0 153.9-103.5 153.9-196.8C384 71.3 298.2 6.5 204 6.5z"
          />
        </svg>
      </a>
    </button>
  </li>
</ul>
`;
